keras/callbacks.py

```diff
@@ -399,7 +399,7 @@ def on_epoch_end(self, epoch, logs=None):
             self.epochs_since_last_save = 0
             filepath = self.filepath.format(epoch=epoch, **logs)
             if self.save_best_only:
-                current = logs.get(self.monitor)
+                current = logs.get(self.monitor)[-1]
```

Comment:
I agree - hence why I mentioned that it not might be the best solution. However, the code as it is does not work at all with temporally-dependent weights!